client/src/components/ContestDetail.js

```diff
@@ -340,7 +359,7 @@ const ContestDetail = ({ classes, auth, contest, match, history }) => {
             </TabPanel>
           </Box>
         </Paper>
-      </Grid>
+      </Grid>: null}
```

Comment:
Not showing the whole component when contest is not loaded yet was done in a more succinct manner when the fetching was done by the other component. A bit unreadable when having null just dangling. Maybe have {!Contest ? null : <REST> } so it's more understandable